### PR TITLE
First version of working test deck.

### PR DESCRIPTION
**Pull request by gerring**

### CODE REVIEW

**Comment by gerring** [comment]
@DiamondLightSource/developers-gda-ispyb-api please review.

**Comment by gerring** [comment]
@mtwharmby and @jamesmudd please take a shifty...

**Comment by gerring** [comment]
The new bits are in ExperimentCommunicationService. The rest of the changes are in reordering the code to stop repeated stuff with the new tests. The new tests are not automated as yet.

File: uk.ac.diamond.ispyb.scanning/src/uk/ac/diamond/ispyb/scanning/ExperimentCommunicationService.java [hunk 1]
```diff
@@ -24,13 +34,77 @@
...
+		return api.retrieveSamplesAssignedForProposal(proposalCode, proposalNumber);
+	}
+
+	public static IspybXpdfFactoryService getIspybXpdfFactoryService() {
+		return ispybXpdfFactoryService;
+	}
+	
+	private static ComponentContext context;
+
+	public void start(ComponentContext c) {
```
Comment (jamesmudd) [inline uk.ac.diamond.ispyb.scanning/src/uk/ac/diamond/ispyb/scanning/ExperimentCommunicationService.java:95]: Instance method sets a static field

File: uk.ac.diamond.ispyb.scanning/src/uk/ac/diamond/ispyb/scanning/ExperimentCommunicationService.java [hunk 2]
```diff
@@ -24,13 +34,77 @@
...
+		context = c;
+	}
+	
+	public void stop() {
```
Comment (jamesmudd) [inline uk.ac.diamond.ispyb.scanning/src/uk/ac/diamond/ispyb/scanning/ExperimentCommunicationService.java:99]: Instance method sets a static field

File: uk.ac.diamond.ispyb.scanning/src/uk/ac/diamond/ispyb/scanning/ExperimentCommunicationService.java [hunk 3]
```diff
@@ -24,13 +34,77 @@
...
+	 * Testing only
+	 * @param api
+	 */
+	public ExperimentCommunicationService(IspybXpdfFactoryService service) {
+		ispybXpdfFactoryService = service;
+	}
+	
+	@Override
+	public synchronized void open() throws SQLException {
+		if (api!=null) throw new IllegalAccessError();
```
Comment (jamesmudd) [inline uk.ac.diamond.ispyb.scanning/src/uk/ac/diamond/ispyb/scanning/ExperimentCommunicationService.java:69]: Might be useful to add a message to this exception like "api is not set"

File: uk.ac.diamond.ispyb.scanning/src/uk/ac/diamond/ispyb/scanning/ExperimentCommunicationService.java [hunk 4]
```diff
@@ -24,13 +34,77 @@
...
+	 */
+	public ExperimentCommunicationService() {
+
+	}
+	
+	/**
+	 * Testing only
+	 * @param api
+	 */
+	public ExperimentCommunicationService(IspybXpdfFactoryService service) {
```
Comment (jamesmudd) [inline uk.ac.diamond.ispyb.scanning/src/uk/ac/diamond/ispyb/scanning/ExperimentCommunicationService.java:63]: Instance method sets a static field

File: uk.ac.diamond.ispyb.scanning/src/uk/ac/diamond/ispyb/scanning/ExperimentCommunicationService.java [hunk 5]
```diff
@@ -107,4 +114,40 @@ public static void setIspybXpdfFactoryService(IspybXpdfFactoryService ispybXpdfF
...
+	}
+
+	@Override
+	public Map<Long, SampleInformation> getSampleInformation(String proposalCode, long proposalNumber, long... sampleIds) {
+		
+		if (sampleIds==null || sampleIds.length<1) throw new IllegalArgumentException("At least one sampleId must be provided");
+		
+		List<Long>   requiredIds = Arrays.stream(sampleIds).boxed().collect(Collectors.toList());
+		List<Sample> samples     = getSamples(proposalCode, proposalNumber);
+		List<Sample> rsamples    = samples.stream().filter(sample->requiredIds.contains(sample.getSampleId())).collect(Collectors.toList());
```
Comment (jamesmudd) [inline uk.ac.diamond.ispyb.scanning/src/uk/ac/diamond/ispyb/scanning/ExperimentCommunicationService.java]: Don't think this one can be shorter as in the filter you are calling 2 instance methods requiredIds.contains and sample.getSampleId

**Review by jamesmudd — COMMENTED** [review]
Only looked at one file ExperimentCommunicationService

**Pull request merged**